hazelcast/src/main/java/com/hazelcast/spi/impl/merge/RingbufferMergeData.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.merge;
```

Comment:
The `RingbufferMergeData` is exposed to the user as `MergingValue.getValue()` and can be created by the user as return value of the merge policy. Is that correct? It should not be in an `impl` then.

Reply:
I'll move it but I'm uncertain about the naming and the API:
- is `RingbufferMergeData` the right name? Can it just be `Ringbuffer` in the merge package? Can it be `MergingRingbuffer`, etc. 
- can we somehow avoid having the capacity as the argument? Does it need to implement `Iterable<E>` or `Collection<E>`? Are some methods missing or are not needed?

Reply:
We already have `Ringbuffer` twice, which is already confusing when picking the right import (maybe the one in `impl` should also be renamed?).

Reply:
Sure, any suggestions? I don't have anything good in mind - `InternalRingbuffer`?